sdks/java/core/src/main/java/org/apache/beam/sdk/runners/inprocess/InProcessSideInputContainer.java

```diff
@@ -220,6 +213,38 @@ public String toString() {
       }
     }
 
+    /**
+     * Gets the future containing the contents of the provided {@link PCollectionView} in the
+     * provided {@link BoundedWindow}, setting up a callback to populate the future with empty
+     * contents if necessary.
+     */
+    private <T> Future<Iterable<? extends WindowedValue<?>>> getViewFuture(
+        final PCollectionView<T> view, final BoundedWindow window)  {
+      PCollectionViewWindow<T> windowedView = PCollectionViewWindow.of(view, window);
+      final SettableFuture<Iterable<? extends WindowedValue<?>>> future =
+          viewByWindows.getUnchecked(windowedView);
+
+      WindowingStrategy<?, ?> windowingStrategy = view.getWindowingStrategyInternal();
+      evaluationContext.scheduleAfterOutputWouldBeProduced(
+          view, window, windowingStrategy, new Runnable() {
+            @Override
+            public void run() {
+              // The requested window has closed without producing elements, so reflect that in
+              // the PCollectionView. If set has already been called, will do nothing.
+              future.set(Collections.<WindowedValue<?>>emptyList());
+        }
+
+        @Override
+        public String toString() {
+          return MoreObjects.toStringHelper("InProcessSideInputContainerEmptyCallback")
```

Comment:
That'll just be some anonymous Runnable. I want the toString to be helpful in figuring out what it's doing.

Done, but via extracting to an inner class.